### PR TITLE
Add-ons screen

### PR DESCRIPTION
**Pull request by stevegrunwell**
This PR leverages the SaaS' `/add-ons` endpoint (see the `feature/add-on-status` branch on that codebase) to show users available add-ons to the base WP101 subscription. This is done through a "card"-style interface (with a bit o' Flexbox to clean it up), linking off to the SaaS to make the actual transaction.

screen shot 2017-12-14 at 5 29 00 pm

Blocked by #4.

### CODE REVIEW

File: views/listings.php [hunk 1]
```diff
@@ -23,7 +25,7 @@
 	</main>
 
 	<nav class="wp101-playlist card">
-		<?php foreach ( $playlist as $series ) : ?>
+		<?php foreach ( $playlist['series'] as $series ) : ?>
```
Comment (bswatson) [inline views/listings.php]: Need to confirm `$playlist['series']` exist before using it in the foreach.

File: views/add-ons.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		<div class="notice notice-warning">
+			<p><?php esc_html_e( 'There are no add-ons currently available for WP101!', 'wp101' ); ?></p>
+		</div>
+
+	<?php else : ?>
+
+		<p><?php esc_html_e( 'Enhance your WP101 experience with these add-ons:', 'wp101' ); ?></p>
+
+		<div class="wp101-addon-list">
+			<?php foreach ( $addons['addons'] as $addon ) : ?>
```
Comment (bswatson) [inline views/add-ons.php:34]: Need to confirm `$addons['addons']` exist before using in the foreach.

File: views/add-ons.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+				<?php $has_addon = in_array( $addon['slug'], $purchased, true ); ?>
```
Comment (bswatson) [inline views/add-ons.php]: Need to very `$addon['slug']` exists before using it.

File: views/add-ons.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+						<h3><?php esc_html_e( 'In this series:', 'wp101' ); ?></h3>
+						<?php TemplateTags\list_topics( $addon['topics'], 3, $addon['url'] ); ?>
+					<?php endif; ?>
+
+					<p class="wp101-addon-button">
+						<?php if ( $has_addon ) : ?>
+							<a href="<?php echo esc_url( admin_url( 'admin.php?page=wp101' ) ); ?>" class="button button-secondary"><?php echo esc_html_e( 'Watch Videos', 'wp101' ); ?></a>
+
+						<?php else : ?>
+							<a href="<?php echo esc_url( $addon['url'] ); ?>" class="button button-primary" target="_blank"><?php echo esc_html_e( 'Get Add-on', 'wp101' ); ?></a>
```
Comment (bswatson) [inline views/add-ons.php]: Should confirm `$addon['url']` exists before using it.

**Review by bswatson — CHANGES_REQUESTED** [review]

File: views/add-ons.php (hunk 2 above)
Comment (stevegrunwell) [reply views/add-ons.php:34]: This check is performed one line 23:

```php
<?php if ( empty( $addons['addons'] ) ) : ?>

  <div class="notice notice-warning">
    <p><?php esc_html_e( 'There are no add-ons currently available for WP101!', 'wp101' ); ?></p>
  </div>

<?php else : ?>

  <p><?php esc_html_e( 'Enhance your WP101 experience with these add-ons:', 'wp101' ); ?></p>

  <!-- ... -->

<?php endif; ?>
```

**Comment by stevegrunwell** [comment]
@bswatson Ready for re-review.

**Review by bswatson — APPROVED** [review]

**Pull request merged**